### PR TITLE
feat: add configuration interface to support customer providers

### PR DESCRIPTION
**Pull request by arbreezy**
Closes # 

## 📑 Description
Refactoring the way we configure k8sgpt ai providers by adding an interface.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

This PR is a prerequisite for adding azure openai and llama providers.

### CODE REVIEW

File: pkg/ai/iai.go [hunk 1]
```diff
@@ -5,12 +5,17 @@ import (
...
 type IAI interface {
-	Configure(token string, model string, language string) error
+	Configure(config IAIConfig, language string) error
 	GetCompletion(ctx context.Context, prompt string) (string, error)
 	Parse(ctx context.Context, prompt []string, nocache bool) (string, error)
 	GetName() string
 }
 
+type IAIConfig interface {
+	GetToken() string
```
Comment (matthisholleville) [inline pkg/ai/iai.go:15]: To be more generic, shouldn't rename this to `GetPassword` ?

File: pkg/ai/prompts.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package ai
```
Comment (matthisholleville) [inline pkg/ai/prompts.go:1]: This is just a suggestion but shouldn't we rename this file to `common.go` ?

**Review by matthisholleville — COMMENTED** [review]
Good job again @arbreezy !

File: pkg/ai/prompts.go (hunk 2 above)
Comment (arbreezy) [reply pkg/ai/prompts.go:1]: I think `prompts.go` is more intuitive tbh if we see a need for more common functionality we rename then

File: pkg/ai/iai.go (hunk 1 above)
Comment (arbreezy) [reply pkg/ai/iai.go]: fair !

**Review by matthisholleville — APPROVED** [review]

**Pull request merged**